### DISCUSSION TITLE
Non-blocking `tab_bar.py`

### DISCUSSION THREAD

**Discussion by jannis-baum** [kovidgoyal/kitty#5076] [Q&A] [answered]
My custom tab bar shows whether the tab's git repository has changes by calling `git status` in a subprocess. `git status` takes its time in very large repositories and blocks/freezes kitty while it loads. Is there any kitty-conventional way of running something asynchronously or loading slow resources to display in the tab bar or should I just use Python's `asyncio` like the `diff`-kitten does for example?

**Comment by kovidgoyal** [comment, marked as answer]
No there isn't. Th tab bar is drawn synchronously. I suggest you run the
subprocess in the background and have it right its result in a text file
that your code reads on next run. Or dont use the pathetically slow git
status use gitstatusd instead. See my watcher repo for an example of
using it.
